Library/brew.sh

```diff
@@ -207,6 +173,56 @@ EOS
   esac
 fi
 
+if [[ -z "$HOMEBREW_DEVELOPER" ]]
+then
+  unset HOMEBREW_RUBY_PATH
+fi
+
+if [[ -z "$HOMEBREW_RUBY_PATH" && "$HOMEBREW_COMMAND" != "install-vendor" ]]
+then
+  HOMEBREW_VENDOR_RUBY_PATH="$HOMEBREW_LIBRARY/Homebrew/vendor/ruby/opt/bin/ruby"
+  if [[ -x "$HOMEBREW_VENDOR_RUBY_PATH" ]]
+  then
+    HOMEBREW_RUBY_PATH="$HOMEBREW_VENDOR_RUBY_PATH"
+  else
+    if [[ -n "$HOMEBREW_OSX" ]]
+    then
+      HOMEBREW_RUBY_PATH="/System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/ruby"
+    else
+      HOMEBREW_RUBY_PATH="$(which ruby)"
```

Comment:
If I recall correctly, we try to avoid this before because user's ruby may be buggy(e.g. jruby). And I think we should only use system ruby or vendor ruby.

Another case we want to avoid this is that, if `which ruby` returns homebrew ruby and users try to `brew reinstall/upgrade ruby` , it will crash.